pkgs/development/interpreters/luajit/default.nix

```diff
@@ -20,24 +20,23 @@ stdenv.mkDerivation rec {
 
   luaversion = "5.1";
 
-  patchPhase = ''
-    substituteInPlace Makefile \
-      --replace /usr/local "$out"
-
-    substituteInPlace src/Makefile --replace gcc cc
-  '' + stdenv.lib.optionalString (stdenv.cc.libc != null)
-  ''
-    substituteInPlace Makefile \
-      --replace ldconfig ${stdenv.cc.libc.bin or stdenv.cc.libc}/bin/ldconfig
+  postPatch = ''
+    substituteInPlace Makefile --replace ldconfig :
   '';
 
   configurePhase = false;
 
+  makeFlags = [
+    "PREFIX=$(out)"
+    "DEFAULT_CC=cc"
+    "CROSS=${stdenv.cc.targetPrefix}"
+    # TODO: when pointer size differs, we would need e.g. -m32
+    "HOST_CC=${buildPackages.stdenv.cc}/bin/cc"
```

Comment:
If `HOST_CC` is the cc for the host platform (the platform the luajit is running on) then this should be:

```
"HOST_CC=${stdenv.cc.targetPrefix}cc"
```

Reply:
However since you also set a cross prefix. They are probably mean the c compiler for the build platform.

Reply:
Their "HOST_CC" is for running the result during build, apparently.  I did test x86_64 -> aarch64 cross, though on 18.09 because our glibc is broken (C.UTF8-related).

Reply:
What is broken with `C.UTF-8`?

Reply:
```
$ nix build -f . --arg crossSystem '(import ./lib).systems.examples.aarch64-multiplatform' glibc
...
  make  -C localedata install-locales
  make[2]: Entering directory '/build/glibc-2.27/localedata'
  .././scripts/mkinstalldirs /nix/store/7lddwdrcjn2c6b5zdwcslv3wpkf3bbfz-glibc-2.27-aarch64-unknown-linux-gnu/lib/locale
  mkdir -p -- /nix/store/7lddwdrcjn2c6b5zdwcslv3wpkf3bbfz-glibc-2.27-aarch64-unknown-linux-gnu/lib/locale
  C.UTF-8.../build/build/elf/ld-linux-aarch64.so.1: line 1: syntax error: unexpected word (expecting ")")
  make[2]: *** [Makefile:212: install-C.UTF-8/UTF-8] Error 2
```
I guess we should move that into a separate thread, though.

Reply:
Yeah some packages consider HOST_CC to be the buildPackages' CC. The "host" in this case corresponds to "build" in Nixpkgs. It's pretty common in Makefiles without autoconf stuff.

Reply:
@vcunat I see. The glibc build process uses its own binaries to build this. I think in our build process we add the build-native library version instead.
I can see if I can imitate this behavior.

Reply:
Or maybe I can just use the C.UTF-8 locale from the build stdenv in the cross-compiling case? Are these files platform dependent?

Reply:
I have no idea.  I can imagine them embedding `size_t` numbers or something...

Reply:
They only have a big/little endian switch as far as I can tell:
https://github.com/NixOS/nixpkgs/pull/56235
I created a patch that fixes the build. Would be great if you could also check the build package.